### DISCUSSION TITLE
Unable to add the host: Cannot find the server resources at http://x.x.x.x

### DISCUSSION THREAD

**Discussion by Gunseerat** [apache/cloudstack#9020] [Q&A] [answered]
##### ISSUE TYPE
 * Other

##### COMPONENT NAME
Cloudstack-agent
~~~

~~~

##### CLOUDSTACK VERSION
4.18

~~~

~~~

##### CONFIGURATION
2 virtual machines setup with management server and KVM host separately.
Both machines configured on NAT Network

##### OS / ENVIRONMENT
Ubuntu 22.04 virtual machines, one configured to run management server and second configured to run KVM host.
Management server has static  IP but no bridge
KVM host has static IP and Bridge cloudbr0

##### SUMMARY
everytime i try adding the host it says unable to add host no server resources found at x.x.x.x

##### STEPS TO REPRODUCE
create two VMs
Add NAT Network
configured management server and KVM host with official documentation
~~~

~~~

##### EXPECTED RESULTS
To be able to add host to management server

~~~

~~~

##### ACTUAL RESULTS
Error in adding host 
Please if there is any workaround using virtual machines and ubuntu... should the cloudbr0 be configured with something else?
~~~

~~~
Logs : Cloudstack agent
● cloudstack-agent.service - CloudStack Agent
 	Loaded: loaded (/lib/systemd/system/cloudstack-agent.service; enabled; vendor preset: enabled)
 	Active: active (running) since Mon 2024-03-25 11:53:12 IST; 25min ago
   	Docs: http://www.cloudstack.org/
   Main PID: 2795 (java)
  	Tasks: 22 (limit: 4531)
 	Memory: 61.7M
    	CPU: 3.072s
 	CGroup: /system.slice/cloudstack-agent.service
         	└─2795 /usr/bin/java -Djava.io.tmpdir=/usr/share/cloudstack-agent/tmp -Xms256m -Xmx2048m -cp "/usr/share/cloudstack-agent/lib/*:/usr/share/cloudstack-agent/plugins/*:/etc/cloudstack/agent:/usr/share/cloudstack-common/scripts" com.cloud.agent.AgentShell

Mar 25 11:53:13 kvm java[2795]:     	at com.cloud.agent.AgentShell.launchNewAgent(AgentShell.java:452)
Mar 25 11:53:13 kvm java[2795]:     	at com.cloud.agent.AgentShell.launchAgentFromClassInfo(AgentShell.java:431)
Mar 25 11:53:13 kvm java[2795]:     	at com.cloud.agent.AgentShell.launchAgent(AgentShell.java:415)
Mar 25 11:53:13 kvm java[2795]:     	at com.cloud.agent.AgentShell.start(AgentShell.java:511)
Mar 25 11:53:13 kvm java[2795]:     	at com.cloud.agent.AgentShell.main(AgentShell.java:541)
Mar 25 11:53:13 kvm java[2795]: Caused by: java.lang.ClassNotFoundException: com.sun.jna.Callback
Mar 25 11:53:13 kvm java[2795]:     	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
Mar 25 11:53:13 kvm java[2795]:     	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
Mar 25 11:53:13 kvm java[2795]:     	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:527)
Mar 25 11:53:13 kvm java[2795]:     	... 18 more

Log : Libvirtd
● libvirtd.service - Virtualization daemon
 	Loaded: loaded (/lib/systemd/system/libvirtd.service; enabled; vendor preset: enabled)
 	Active: active (running) since Mon 2024-03-25 11:52:35 IST; 25min ago
   	Docs: man:libvirtd(8)
         	https://libvirt.org
   Main PID: 2505 (libvirtd)
  	Tasks: 21 (limit: 32768)
 	Memory: 18.2M
    	CPU: 355ms
 	CGroup: /system.slice/libvirtd.service
         	├─1037 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/default.conf --leasefile-ro --dhcp-script=/usr/lib/libvirt/libvirt_leaseshelper
         	├─1038 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/default.conf --leasefile-ro --dhcp-script=/usr/lib/libvirt/libvirt_leaseshelper
         	└─2505 /usr/sbin/libvirtd

Mar 25 11:52:35 kvm systemd[1]: Starting Virtualization daemon...
Mar 25 11:52:35 kvm systemd[1]: Started Virtualization daemon.
Mar 25 11:52:36 kvm dnsmasq[1037]: read /etc/hosts - 9 addresses
Mar 25 11:52:36 kvm dnsmasq[1037]: read /var/lib/libvirt/dnsmasq/default.addnhosts - 0 addresses
Mar 25 11:52:36 kvm dnsmasq-dhcp[1037]: read /var/lib/libvirt/dnsmasq/default.hostsfile
Mar 25 11:52:36 kvm libvirtd[2505]: libvirt version: 8.0.0, package: 1ubuntu7.8 (Lena Voytek <lena.voytek@canonical.com> Wed, 29 Nov 2023 14:52:52 -0700)
Mar 25 11:52:36 kvm libvirtd[2505]: hostname: kvm
Mar 25 11:52:36 kvm libvirtd[2505]: Configured security driver "none" disables default policy to create confined guests

Log: Management Server
● cloudstack-management.service - CloudStack Management Server
 	Loaded: loaded (/lib/systemd/system/cloudstack-management.service; enabled; vendor preset: enabled)
 	Active: active (running) since Mon 2024-03-25 14:00:12 IST; 10min ago
   Main PID: 2566 (java)
  	Tasks: 183 (limit: 4598)
 	Memory: 1.0G
    	CPU: 1min 51.447s
 	CGroup: /system.slice/cloudstack-management.service
         	└─2566 /usr/bin/java -Djava.security.properties=/etc/cloudstack/management/java.security.ciphers -Djava.awt.headless=true -Dcom.sun.management.jmxremote=false -Xmx2G -XX:+UseParallelGC -XX:MaxGCPauseMillis=500 -XX:+HeapDumpOn>

Mar 25 14:07:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-3:ctx-bc3ef61c) (logid:a6d0aea0) used memory from /proc: 1106956
Mar 25 14:08:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-4:ctx-5b01a77a) (logid:03aab6d7) system memory from /proc: 4097032192
Mar 25 14:08:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-4:ctx-5b01a77a) (logid:03aab6d7) free memory from /proc: 152035328
Mar 25 14:08:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-4:ctx-5b01a77a) (logid:03aab6d7) used memory from /proc: 1107212
Mar 25 14:09:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-1:ctx-664f53b5) (logid:ad8b2476) system memory from /proc: 4097032192
Mar 25 14:09:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-1:ctx-664f53b5) (logid:ad8b2476) free memory from /proc: 134844416
Mar 25 14:09:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-1:ctx-664f53b5) (logid:ad8b2476) used memory from /proc: 1117452
Mar 25 14:10:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-3:ctx-bc7de9ee) (logid:dcda9e97) system memory from /proc: 4097032192
Mar 25 14:10:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-3:ctx-bc7de9ee) (logid:dcda9e97) free memory from /proc: 121516032
Mar 25 14:10:04 cloud java[2566]: INFO  [c.c.s.StatsCollector] (StatsCollector-3:ctx-bc7de9ee) (logid:dcda9e97) used memory from /proc: 1112624
lines 1-20/20 (END)

**Comment by rajujith** [comment]
@Gunseerat  can you check whether you have selected the hypervisor type as KVM while adding cluster and host?

**Comment by weizhouapache** [comment]
@Gunseerat 
do you use a non-root user ?

**Comment by Gunseerat** [comment]
@rajujith yes sir the hypervisor is kvm selected while creating host
@weizhouapache I have added all permissions in sudoers file also

**Comment by Gunseerat** [comment]
I do not understand what server resources is it referencing to...

**Comment by weizhouapache** [comment]
> @rajujith yes sir the hypervisor is kvm selected while creating host @weizhouapache I have added all permissions in sudoers file also

@Gunseerat 
and wheel group as well ?
refer to https://github.com/apache/cloudstack/issues/8851#issuecomment-2030514313

**Comment by yadvr** [comment, marked as answer]
same as https://github.com/apache/cloudstack/issues/8851 - looks like kvm setup issue.
